File: lessons/week_1/assignments/terminalClock.go
package main

import (
	"fmt"
	"time"
)

// clearTerminal clears the terminal
func clearTerminal() {
	// ANSI escape code to clear the terminal and move the cursor to the top-left corner
	fmt.Print("\033[H\033[2J")
}

func main() {
	for {
		clearTerminal()

		var hour = time.Now().Hour()
		var minutes = time.Now().Minute()
		var seconds = time.Now().Second()

		var timeNow = fmt.Sprintf("%02d:%02d:%02d", hour, minutes, seconds)

		fmt.Println("The time is: ", timeNow)

		time.Sleep(time.Second)
	}

}

File: lessons/week_1/assignments/terminalClock.js
setInterval(() => {
    console.clear();
    
    let currentTime = new Date();
    let hours = currentTime.getHours()
    let minutes = currentTime.getMinutes()
    let seconds = currentTime.getSeconds()

    if (hours < 10) {
        hours = `0${hours}`;
    } else if (minutes < 10) {
        minutes = `0${minutes}`;
    } else if (seconds < 10) {
        seconds = `0${seconds}`;
    }

    console.log(`${hours}:${minutes}:${seconds}`)

}, 1 * 1000)